COCOAPODS.md

```diff
@@ -1,150 +1,379 @@
+    }
+    ```
+    
+    </div>
 
-## Workflow
+2. Re-import the project.
 
-To import a Kotlin/Native module in an existing Xcode project:
+If you want to use dependencies on local pods from Kotlin code, import the corresponding packages.
 
-0. Make sure that you have CocoaPods [installed](https://guides.cocoapods.org/using/getting-started.html#installation).
-We recommend using CocoaPods 1.6.1 or later.
+<div class="sample" markdown="1" theme="idea" data-highlight-only>
+
+```kotlin
+import cocoapods.pod_dependency.*
+import cocoapods.subspec_dependency.*
+```
 
-1. Configure a Gradle project: apply the `org.jetbrains.kotlin.native.cocoapods` plugin, add
-and configure the targets, and specify the required podspec fields.
+</div>
 
-2. Run the `podspec` task. The podspec file described above will be generated.
+You can find a sample project [here](https://github.com/zoldater/KotlinWithLocalObjCPods). 
+ 
+## Use a Kotlin Gradle project as a CocoaPods dependency
 
-    > In order to avoid compatibility issues during an Xcode build, the plugin requires using
-    a [Gradle wrapper](https://docs.gradle.org/current/userguide/gradle_wrapper.html). 
-    To generate the wrapper automatically during execution of the `podspec` task,
-    run it with the parameter `-Pkotlin.native.cocoapods.generate.wrapper=true`.
+You can use a multiplatform project with native targets as a CocoaPods dependency (Kotlin Pod). You can include such a dependency
+  in a `Podfile` of an Xcode project  by its name and path to the project directory containing the generated `Podspec`. 
+This dependency will be automatically built (and rebuilt) along with this project. 
+Such approach simplifies importing to Xcode by removing a need to write corresponding Gradle tasks and Xcode build steps manually.
+ 
+You can add dependencies between:
+* [A Kotlin Pod and an Xcode project with one target](#add-a-dependency-between-a-kotlin-pod-and-xcode-project-with-one-target)
+* [A Kotlin Pod and an Xcode project with several targets](#add-a-dependency-between-a-kotlin-pod-with-an-xcode-project-with-several-targets)
+* [Several Kotlin Pods and an Xcode project](#add-dependencies-between-multiple-kotlin-pods-and-an-xcode-project)
 
-3. Add a reference to the generated podspec in a Podfile of the Xcode project.
+> To correctly import the dependencies into the Kotlin/Native module, the
+`Podfile` must contain either [`use_modular_headers!`](https://guides.cocoapods.org/syntax/podfile.html#use_modular_headers_bang)
+or [`use_frameworks!`](https://guides.cocoapods.org/syntax/podfile.html#use_frameworks_bang)
+directive.
+{:.note}
+
+### Add a dependency between a Kotlin Pod and Xcode project with one target
+
+1. Create an Xcode project with `Podfile` if you haven’t done this yet.
+2. Add the path to your `Podfile` with `podfile = project.file(..)` to `build.gradle.kts` (`build.gradle`) 
+of your Kotlin project.  
+    This step helps synchronize an Xcode project and Kotlin Pod dependencies.
```

Comment:
May be explicitly say that this synchronization is performed by calling `pod install` for the user's Xcode project? 

I mean a user may add some third-party dependencies to Podfile, re-import the Gradle project to IDEA and unexpectedly get Xcode dependencies updated too. It may be a WTF for the user so it worth warning him. I think that just saying that `pod install` is called on IDEA re-import is enough.

Reply:
Updated the explanation as discussed with @zoldater.